warehouse/query-core/src/main/java/datawave/query/config/ShardQueryConfiguration.java

```diff
@@ -451,6 +451,12 @@ public class ShardQueryConfiguration extends GenericQueryConfiguration implement
      */
     private boolean pruneQueryOptions = false;
 
+    /**
+     * The maximum number of lines to print when streaming the query from the global index. Useful for limiting the logging footprint of large queries when
+     * debug logging is enabled.
+     */
+    private int maxLinesToPrint = -1;
```

Comment:
Consider a slightly more descriptive variable name? (e.g., maxDebugLogLines)

Reply:
I'm not certain that being more descriptive is better in this case. Right now this mostly affects log messages at the debug level, but this could (and should) be used to drive the number of lines printed elsewhere in query planning. Those other instances may be at the info or warn levels.